builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -373,6 +384,12 @@ func resourceVSphereVirtualMachineCreate(d *schema.ResourceData, meta interface{
 		memoryMb: int64(d.Get("memory").(int)),
 	}
 
+	if v, ok := d.GetOk("memory_reservation"); ok {
```

Comment:
if we need to do this logic, shouldn't we just set a Default of 0 in the parameter and then document that? It would mean we only even just have to do this

```
vm.memoryAllocation.reservation = int64(v.(int))
```